### PR TITLE
Port fixes

### PR DESCRIPTION
**Pull request by mjeanson**

### CODE REVIEW

File: cli/babeltrace.c [hunk 1]
```diff
@@ -2584,7 +2584,7 @@ void set_auto_log_levels(struct bt_config *cfg)
 				 * explicitly specified.
 				 */
 				val[0] = cfg->log_level;
-				setenv(*env_var_name, val, 1);
+				g_setenv(*env_var_name, val, 1);
```
Comment (jgalar) [inline cli/babeltrace.c]: I made the change in other places:

```
diff --git a/include/babeltrace/compat/utc-internal.h b/include/babeltrace/compat/utc-internal.h
index d78a1885..db3035a3 100644
--- a/include/babeltrace/compat/utc-internal.h
+++ b/include/babeltrace/compat/utc-internal.h
@@ -46,6 +46,7 @@ time_t bt_timegm(struct tm *tm)
 
 #include <string.h>
 #include <stdlib.h>
+#include <glib.h>
 
 /*
  * Note: Below implementation of timegm() is not thread safe
@@ -76,11 +77,11 @@ time_t bt_timegm(struct tm *tm)
        }
 
        /* Temporarily setting TZ to 1 for UTC */
-       setenv("TZ", "", 1);
+       g_setenv("TZ", "", 1);
        tzset();
        ret = mktime(tm);
        if (tz) {
-               setenv("TZ", tz, 1);
+               g_setenv("TZ", tz, 1);
                free(tz);
        } else {
                unsetenv("TZ");
diff --git a/tests/lib/test-plugin-plugins/minimal.c b/tests/lib/test-plugin-plugins/minimal.c
index c78862b0..6dec80aa 100644
--- a/tests/lib/test-plugin-plugins/minimal.c
+++ b/tests/lib/test-plugin-plugins/minimal.c
@@ -20,13 +20,13 @@
 
 static enum bt_plugin_status plugin_init(struct bt_plugin *plugin)
 {
-       setenv("BT_TEST_PLUGIN_INIT_CALLED", "1", 1);
+       g_setenv("BT_TEST_PLUGIN_INIT_CALLED", "1", 1);
        return BT_PLUGIN_STATUS_OK;
 }
 
 static enum bt_plugin_status plugin_exit(void)
 {
-       setenv("BT_TEST_PLUGIN_EXIT_CALLED", "1", 1);
+       g_setenv("BT_TEST_PLUGIN_EXIT_CALLED", "1", 1);
        return BT_PLUGIN_STATUS_OK;
 }
 
diff --git a/tests/lib/test_plugin.c b/tests/lib/test_plugin.c
index 874a11db..1fb3c6dd 100644
--- a/tests/lib/test_plugin.c
+++ b/tests/lib/test_plugin.c
@@ -49,8 +49,8 @@ static int check_env_var(const char *name)
 
 static void reset_test_plugin_env_vars(void)
 {
-       setenv("BT_TEST_PLUGIN_INIT_CALLED", "0", 1);
-       setenv("BT_TEST_PLUGIN_EXIT_CALLED", "0", 1);
+       g_setenv("BT_TEST_PLUGIN_INIT_CALLED", "0", 1);
+       g_setenv("BT_TEST_PLUGIN_EXIT_CALLED", "0", 1);
 }
 
 static char *get_test_plugin_path(const char *plugin_dir,
```

File: compat/compat_mman.c [hunk 2]
```diff
@@ -29,6 +29,9 @@
 #define BT_LOG_TAG "COMPAT-MMAN"
 #include "logging.h"
 
+BT_HIDDEN
+int bt_mman_dummy_symbol;
```
Comment (jgalar) [inline compat/compat_mman.c]: why is this needed?

File: logging/log.c [hunk 3]
```diff
@@ -305,6 +305,7 @@
 	#include <sys/time.h>
 	#if defined(__linux__)
 		#include <linux/limits.h>
+	#elif (defined(__sun__) || defined(__CYGWIN__))
```
Comment (jgalar) [inline logging/log.c]: Added `/* Solaris and Cygwin have no sys/syslimits.h */`

File: compat/compat_mman.c (hunk 2 above)
Comment (mjeanson) [inline compat/compat_mman.c]: The macOS linker doesn't like empty object files, could be ifdefed to __APPLE__

Comment (jgalar) [inline compat/compat_mman.c]: Good idea. I'd say ifdef to APPLE and add a comment explaining why that's needed.

**Comment by jgalar** [comment]
Merged, thanks!
